### ISSUE TITLE
Behavior: Arrow Key Focus

### ISSUE THREAD

**Issue by T-Hugs** [primer/components#1134]
## Description
We need a behavior that sets up focus management with arrow keys (and potentially other keys). According to [ARIA Best Practices](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_generalnav), there are many cases where focus should be moved using arrow keys, but the DOM only has native support for managing focus using the Tab key.

Furthermore, we should support the [`aria-activedescendant` pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_focus_activedescendant), which will be used in "filter list" UI and potentially other situations.

Due to the complexity of this behavior, it should be written in "vanilla" JS and wrapped by a hook.

### Specification & Usage
See the [proposed documentation](https://primer-components-f5ts87di8-primer.vercel.app/components/focusZone) for more details.
